### PR TITLE
feat: enable loading all the cryptokitties assets with recursion

### LINKED ISSUE #807: UI is only getting first 20 cryptokitties due to API limits
we need to use pagination to get other cryptokitties assets

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #807 
Before, we only loaded 20 because that is the Cryptokitties API limit.
Now, we load everything using a recursive function and increasing the offset

### CODE REVIEW

**Comment by michaelsbradleyjr** [comment]
Is it possible to get a stack overflow if the recursion goes too deep?

**Comment by jrainville** [comment]
> Is it possible to get a stack overflow if the recursion goes too deep?

Hmm, good question. I don't know what the limit, but I'll try adding the biggest kitty holder as a watch only address

**Comment by jrainville** [comment]
I found that some users have a quantity of kitties that is excessively high, so for now, we decided to just limit the amount be show.

I added a warning as well.

File: src/status/wallet/collectibles.nim [hunk 1]
```diff
@@ -56,47 +58,58 @@ proc tokensOfOwnerByIndex(contract: Contract, address: EthAddress): seq[int] =
...
+  let response = client.request(url)
+  let responseBody = parseJson(response.body)
+  let kitties = responseBody["kitties"]
+  for kitty in kitties:
+    try:
+      var id = kitty["id"]
+      var name = kitty["name"]
+      var finalId = ""
+      var finalName = ""
+      if (not (id.kind == JNull)):
```
Comment (richard-ramos) [inline src/status/wallet/collectibles.nim:83]:
```suggestion
      if id.kind != JNull:
```

File: src/status/wallet/collectibles.nim [hunk 2]
```diff
@@ -56,47 +58,58 @@ proc tokensOfOwnerByIndex(contract: Contract, address: EthAddress): seq[int] =
...
+        finalId = $id
+      if (not (name.kind == JNull)):
```
Comment (richard-ramos) [inline src/status/wallet/collectibles.nim:85]:
```suggestion
      if name.kind != JNull:
```

File: src/status/wallet/collectibles.nim [hunk 3]
```diff
@@ -56,47 +58,58 @@ proc tokensOfOwnerByIndex(contract: Contract, address: EthAddress): seq[int] =
...
+        finalName = $name
+      cryptokitties.add(Collectible(id: finalId,
+      name: finalName,
+      image: kitty["image_url_png"].str,
+      collectibleType: CRYPTOKITTY,
+      description: "",
+      externalUrl: ""))
```
Comment (richard-ramos) [inline src/status/wallet/collectibles.nim:92]:
```suggestion
        name: finalName,
        image: kitty["image_url_png"].str,
        collectibleType: CRYPTOKITTY,
        description: "",
        externalUrl: ""))
```

**Review by richard-ramos — APPROVED** [review]
Good job!

File: src/status/wallet/collectibles.nim [hunk 4]
```diff
@@ -56,47 +58,58 @@ proc tokensOfOwnerByIndex(contract: Contract, address: EthAddress): seq[int] =
...
-      except Exception as e2:
-        error "Error with this individual cat", msg = e2.msg, cat = kitty
+    except Exception as e2:
+      error "Error with this individual cat", msg = e2.msg, cat = kitty
+
+  let limit = responseBody["limit"].getInt
+  let total = responseBody["total"].getInt
+  let currentCount = limit * (offset + 1)
+  if (currentCount < total and currentCount < MAX_TOKENS):
+    # Call the API again with oofset + 1
```
Comment (emizzle) [inline src/status/wallet/collectibles.nim:100]:
```suggestion
    # Call the API again with offset + 1
```

File: ui/app/AppLayouts/Wallet/components/collectiblesComponents/CollectiblesHeader.qml [hunk 5]
```diff
@@ -29,10 +29,23 @@ Rectangle {
...
         text: collectibleHeader.collectibleName
         anchors.left: collectibleIconImage.right
         anchors.leftMargin: Style.current.padding
         anchors.verticalCenter: parent.verticalCenter
+        font.pixelSize: 17
+    }
+
+    StyledText {
+        visible: collectiblesQty >= Constants.maxTokens
+        text: qsTr("Maximum number of collectibles to display reached")
```
Comment (emizzle) [inline ui/app/AppLayouts/Wallet/components/collectiblesComponents/CollectiblesHeader.qml:42]:
```suggestion
        text: qsTr("Maximum number of collectibles to display reached. You have waaaaaay too many kitties. Please find something else to do 😝")
```
😜

**Review by emizzle — APPROVED** [review]
Nice work!

File: ui/app/AppLayouts/Wallet/components/collectiblesComponents/CollectiblesHeader.qml (hunk 5 above)
Comment (jrainville) [reply ui/app/AppLayouts/Wallet/components/collectiblesComponents/CollectiblesHeader.qml:42]: 😂

**Comment by jrainville** [comment]
Rebased

**Pull request merged**